APNGImageSerialization/Classes/APNGImageSerialization.m

```diff
@@ -247,10 +252,44 @@ + (NSData *)dataWithImages:(NSArray<UIImage *> *)images
 
 @end
 
+#pragma mark -
+
+#ifndef ANIMATED_PNG_NO_UIIMAGE_INITIALIZER_SWIZZLING
+#import <objc/runtime.h>
+
+static inline void apng_swizzleSelector(Class class, SEL originalSelector, SEL swizzledSelector) {
+    Method originalMethod = class_getInstanceMethod(class, originalSelector);
+    Method swizzledMethod = class_getInstanceMethod(class, swizzledSelector);
+    if (class_addMethod(class, originalSelector, method_getImplementation(swizzledMethod), method_getTypeEncoding(swizzledMethod))) {
+        class_replaceMethod(class, swizzledSelector, method_getImplementation(originalMethod), method_getTypeEncoding(originalMethod));
+    } else {
+        method_exchangeImplementations(originalMethod, swizzledMethod);
+    }
+}
+
+@interface UIImage (Animated_PNG)
+@end
 
 @implementation UIImage (Animated_PNG)
 
-+ (UIImage *)animatedImageNamed:(NSString *)name
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        @autoreleasepool {
+            apng_swizzleSelector(object_getClass((id)self), @selector(imageNamed:), @selector(apng_animatedImageNamed:));
```

Comment:
In fact, you have replaced `imageNamed:` method here.

Reply:
I did originally, but then made changes in subsequent commits to not swizzle that function. If you view all changes it will not be swizzled